### PR TITLE
feat: add jss

### LINKED ISSUE #17: Lets add JSS
- [ ] jss
- [ ] react-jss
- [ ] styled-jss

### PR DESCRIPTION
**Pull request by DSchau**
@kof would you want to take a look at this?

I really, really wanted to use react-jss, or styled-jss, but I was getting all kinds of errors. Would appreciate any help, the basic structure of how you add a library should be demonstrated by this commit!

Fixes #17

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "glamorous": "^4.7.0",
     "history": "^4.6.3",
     "jss": "^8.1.0",
+    "jss-extend": "^5.0.0",
```
Comment (kof) [inline package.json]: you don't need that, it is already part of the preset

File: src/snippets/jss.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
+import React, { Component } from 'react';
+import jss from 'jss';
+import preset from 'jss-preset-default';
+import extend from 'jss-extend';
+import Logo from './logo';
+
+jss.setup(preset());
+jss.setup(extend());
```
Comment (kof) [inline src/snippets/jss.js]: already part of the preset

**Comment by kof** [comment]
Can you describe errors you had? I would like to fix them if they are caused by the library.

File: src/snippets/jss.js (hunk 2 above)
Comment (DSchau) [reply src/snippets/jss.js]: Will fix!

**Comment by DSchau** [comment]
@kof I just committed the [styled-jss branch](https://github.com/DSchau/css-in-js-playground/compare/styled-jss?expand=1). If you'd like you can check it out yourself and see the issues.

My first instinct is that I need to configure a Webpack loader differently, _or_ it could be caused by using React 16. Here is the (first) error:

```
Uncaught (in promise) Error: Cannot find module "react-dom/lib/DOMProperty"
    at webpackMissingModule (index.js:7)
    at Object.398 (index.js:7)
    at __webpack_require__ (bootstrap fb3149dccdb8137fc7cf:705)
    at fn (bootstrap fb3149dccdb8137fc7cf:111)
    at Object.397 (filterProps.js:7)
    at __webpack_require__ (bootstrap fb3149dccdb8137fc7cf:705)
    at fn (bootstrap fb3149dccdb8137fc7cf:111)
    at Object.396 (styled.js:11)
    at __webpack_require__ (bootstrap fb3149dccdb8137fc7cf:705)
    at fn (bootstrap fb3149dccdb8137fc7cf:111)
```

and the second

```
client:139 ./node_modules/is-react-prop/lib/index.js
Module not found: Error: Can't resolve 'react-dom/lib/EventPluginRegistry' in '/Users/dschau/Code/Personal/css-in-js/playground/node_modules/is-react-prop/lib'
 @ ./node_modules/is-react-prop/lib/index.js 11:27-71
 @ ./node_modules/styled-jss/utils/filterProps.js
 @ ./node_modules/styled-jss/styled.js
 @ ./node_modules/styled-jss/createStyled.js
 @ ./node_modules/styled-jss/index.js
 @ ./src/utils/libraries.ts
 @ ./src/components/CodePreview/index.tsx
 @ ./src/components/CodeProvider/index.tsx
 @ ./src/App.tsx
 @ ./src/index.tsx
 @ multi (webpack)-dev-server/client?http://0.0.0.0:8000 webpack/hot/dev-server ./src/index
```

**Comment by kof** [comment]
Oh yeah, thats the issue with styled-jss, it is using a package named is-react-prop, which relies on react's internals. Lets skip styled-jss for now.

cc @lttb

**Comment by DSchau** [comment]
@kof react-jss should be good though? I thought I tried that one too with similar results, but I'm down to give it another go.

**Comment by kof** [comment]
react-jss should work without issues

**Comment by lttb** [comment]
> Oh yeah, thats the issue with styled-jss, it is using a package named is-react-prop, which relies on react's internals. Lets skip styled-jss for now.

@DSchau @kof right, `is-react-prop` is incompatible with React 16 now, but I hope I'll fix it this weekend

File: src/snippets/jss.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    alignItems: 'center',
+    height: 60,
+    width: 60,
+    borderRadius: 60,
+    backgroundColor: 'white',
+    boxShadow: { x: 0, y: 1, blur: 6, color: 'rgba(0, 0, 0, 0.1)' },
+    position: 'absolute',
+    top: '1rem',
+    left: '1rem',
+    zIndex: '2'
```
Comment (kof) [inline src/snippets/jss.js:37]: btw. you can use a number here

Comment (DSchau) [reply src/snippets/jss.js:37]: Will fix :) Good catch, and thank you again for the help!

Comment (DSchau) [reply src/snippets/jss.js:37]: Feel free to open up an issue (or submit a PR :) ) when styled-jss fixes the React 16 issues!

**Pull request merged**